### PR TITLE
Split issue templates by categories

### LINKED ISSUE #1639: Split issue templates by categories
Our Github [issue template](https://github.com/pydanny/cookiecutter-django/blob/e4d94b52d5aa0007e8def914d69b300249525765/.github/ISSUE_TEMPLATE.md) is a good way to ask people the questions we need to resolve their issue, but depending on what is being submitted, some questions are not necessarily relevant.

The Github feature has been improved recently to allow having multiple templates: https://blog.github.com/2018-01-25-multiple-issue-and-pull-request-templates/

[Jest](https://github.com/facebook/jest/) implemented that in their repository if you're curious to see how it looks.

We could have 3 templates:
- Bug report: step to reproduce, expected behaviour, actual behaviour, environment details
- New feature request: motivation for the new feature, expected behaviour
- Support Request: channels to get support (stackoverflow, gitter?,...), email address for paid support, links to foundraising pages

I'm hoping that having a smaller, more appropriate template would reduce the number of report with incomplete information.

### PR DESCRIPTION
**Pull request by webyneter**
* [x] Bug Report template
* [x] Feature Request template
* [x] Improvement Suggestion template
* [x] Paid Support template
* [x] Question template
* [x] Regression Report template
* [x] Update root ISSUE_TEMPLATE
* <s> ? [CODEOWNERS](https://help.github.com/articles/about-codeowners/)</s>
* [x] [PULL_REQUEST_TEMPLATE](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/)

Closes #1639.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/question.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+Before raising a question, make sure to
+
+* Examine [the docs](https://cookiecutter-django.readthedocs.io/en/latest/)
+* Check out [Gitter](https://gitter.im/pydanny/cookiecutter-django)
+* Check out [StackOverflow](https://stackoverflow.com/questions/tagged/cookiecutter-django)
+
+
+## The question
```
Comment (browniebroke) [inline .github/ISSUE_TEMPLATE/question.md]: The old template said: "for support questions, please use the `cookiecutter-django` tag on stackoverflow", I think we should keep directing people there.

File: .github/ISSUE_TEMPLATE/improvement.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+name: Improvement Suggestion
```
Comment (browniebroke) [inline .github/ISSUE_TEMPLATE/improvement.md]: This looks the same as the "feature" template. While the difference between improvement and feature is subtle, it's not worth making a different template for it, IMO. 
I don't feel strongly about it, though. Feel free to keep it if you do.

Comment (webyneter) [reply .github/ISSUE_TEMPLATE/improvement.md]: The way I see it, a "feature" is something new while an "improvement" is, well, an improvement to something that already exists.

File: .github/ISSUE_TEMPLATE/question.md (hunk 1 above)
Comment (webyneter) [reply .github/ISSUE_TEMPLATE/question.md]: > for support questions, please use the cookiecutter-django tag on stackoverflow

always sounded a bit ambigous to me: why would people still come and create tickets here even though other channels exist? To my mind, everyone would be better off expressing their concerns via a single conventional channel -- Issues, in that case. 
I might be missing on something here, of course...

File: .github/ISSUE_TEMPLATE/improvement.md (hunk 2 above)
Comment (webyneter) [reply .github/ISSUE_TEMPLATE/improvement.md]: I always thoght of "Feature Requests" as suggestions of new functionality while Improvements are suggestions about, well, improving existing features...

Comment (webyneter) [reply .github/ISSUE_TEMPLATE/improvement.md]: But, yes, they do represent related concepts in the end.

Comment (browniebroke) [reply .github/ISSUE_TEMPLATE/improvement.md]: We can try with both. It'll be easy to remove if a wrong one gets picked too often.

Comment (webyneter) [reply .github/ISSUE_TEMPLATE/improvement.md]: I agree.

File: .github/ISSUE_TEMPLATE/question.md (hunk 1 above)
Comment (browniebroke) [reply .github/ISSUE_TEMPLATE/question.md]: The purpose of the other channels is that they require less bandwidth from the maintainers, anyone can answer questions on Stackoverflow or Gitter without us having to triage a record on Github. 

Also, having a question about cookiecutter-django often actually means a question about Django itself or a library we depend upon. Stackoverflow is a better channel for that type of things, where other people can help out.

Comment (webyneter) [reply .github/ISSUE_TEMPLATE/question.md]: Yes, you've got a point.

**Comment by webyneter** [comment]
@browniebroke what do you think about the PR template?

**Pull request merged**